src/main/java/gim/model/exercise/ExerciseList.java

```diff
@@ -3,78 +3,55 @@
 
-//import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 
-//import gim.model.exercise.exceptions.DuplicateExerciseException;
 import gim.model.exercise.exceptions.ExerciseNotFoundException;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 
 /**
- * A list of exercises that enforces uniqueness between its elements and does not allow nulls.
- * A exercise is considered unique by comparing using {@code Exercise#isSameExercise(Exercise)}. As such,
- * adding and updating of exercises uses Exercise#isSameExercise(Exercise) for equality so as to ensure that the
- * exercise being added or updated is unique in terms of identity in the UniqueExerciseList. However, the removal of an
- * exercise uses Exercise#equals(Object) so as to ensure that the exercise with exactly the same fields will be removed.
+ * A list of exercises that does not allow nulls.
  *
  * Supports a minimal set of list operations.
  *
- * @see Exercise#isSameExercise(Exercise)
  */
-public class UniqueExerciseList implements Iterable<Exercise> {
+public class ExerciseList implements Iterable<Exercise> {
 
     private final ObservableList<Exercise> internalList = FXCollections.observableArrayList();
     private final ObservableList<Exercise> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList);
-    //     private final HashMap<Name, Exercise> test;
-
-    //     public UniqueExerciseList() {
-    //         test = new HashMap<>();
-    //     }
 
     /**
      * Returns true if the list contains an equivalent exercise as the given argument.
      */
     public boolean contains(Exercise toCheck) {
         requireNonNull(toCheck);
-        //        return test.containsKey(toCheck.getName());
         return internalList.stream().anyMatch(toCheck::isSameExercise);
     }
 
     /**
-     * Adds an exercise to the list.
-     * The exercise must not already exist in the list.
+     * Adds an Exercise to the List and exerciseHashMap.
+     * If the Exercise already exists, i.e. two Exercises with the same Name, categorise them together.
      */
     public void add(Exercise toAdd) {
         requireNonNull(toAdd);
-        //        if (!contains(toAdd)) {
-        //            test.put(toAdd.getName(), toAdd);
-        //        }
-        //        if (contains(toAdd)) {
-        //            throw new DuplicateExerciseException();
-        //        }
         internalList.add(toAdd);
     }
 
     /**
-     * Replaces the exercise {@code target} in the list with {@code editedExercise}.
+     * Replaces the Exercise {@code target} in the list with {@code editedExercise}.
      * {@code target} must exist in the list.
-     * The exercise identity of {@code editedExercise} must not be the same as another existing exercise in the list.
      */
     public void setExercise(Exercise target, Exercise editedExercise) {
+        // Might be removing this?
```

Comment:
Is this needed?

Reply:
Left in for now, will remove in subsequent PRs under the issue 'Remove EditExercise'.